python/chronos/src/bigdl/chronos/autots/model/base_automodel.py

```diff
@@ -318,3 +316,33 @@ def get_best_config(self):
 
     def _get_best_automl_model(self):
         return self.best_model
+
+
+def Dynamic_binding(cls):
+    super_init = cls.__init__
+    super_call = cls.__call__
+
+    def __init__(cls, *args, **kwag):
+        super_init(cls, *args, **kwag)
+        super_call(cls, *args, **kwag)
+
+        # dynamic binding
+        if cls.backend.startswith("torch"):
+            from bigdl.orca.automl.model.base_pytorch_model import PytorchModelBuilder
+            cls._DEFAULT_BEST_MODEL_DIR = "best_model.ckpt"
+            cls._DEFAULT_BEST_CONFIG_DIR = "best_config.json"
+            model_builder = PytorchModelBuilder(model_creator=cls._model_creator,
+                                                optimizer_creator=cls.optimizer,
+                                                loss_creator=cls.loss)
+        elif cls.backend.startswith("keras"):
+            from bigdl.orca.automl.model.base_keras_model import KerasModelBuilder
+            cls.search_space.update({"optimizer": cls.optimizer, "loss": cls.loss})
+            model_builder = KerasModelBuilder(model_creator=cls._model_creator)
+            cls._DEFAULT_BEST_MODEL_DIR = "best_keras_model.ckpt"
+            cls._DEFAULT_BEST_CONFIG_DIR = "best_keras_config.json"
+
+        from bigdl.orca.automl.auto_estimator import AutoEstimator
+        cls.auto_est = AutoEstimator(model_builder, **cls.auto_est_config)
+
+    cls.__init__ = __init__
+    return cls
```

Comment:
I am not quite sure why we need this decorator. It seems much easier if we can just move this to baseautomodel's `__init__`.